packages/flutter_feed_utils/lib/src/widgets/overview_layout.dart

```diff
@@ -0,0 +1,81 @@
+import "package:flutter/cupertino.dart";
+import "package:flutter/material.dart";
+import "package:flutter/widgets.dart";
+
+enum _OverviewLayoutOption {
+  list,
+  grid;
+}
+
+class OverviewLayout extends StatelessWidget {
+  const OverviewLayout._({
+    required _OverviewLayoutOption activeLayout,
+    required this.items,
+    this.columnCount = 1,
+  }) : _activeLayout = activeLayout;
+
+  factory OverviewLayout.list({required List<Widget> children}) =>
+      OverviewLayout._(
+        activeLayout: _OverviewLayoutOption.list,
+        items: children,
+      );
+
+  factory OverviewLayout.grid({
+    required int columnCount,
+    required List<Widget> children,
+  }) =>
+      OverviewLayout._(
+        activeLayout: _OverviewLayoutOption.grid,
+        columnCount: columnCount,
+        items: children,
+      );
+
+  final List<Widget> items;
+  final _OverviewLayoutOption _activeLayout;
+  final int columnCount;
+
+  @override
+  Widget build(BuildContext context) => switch (_activeLayout) {
+        _OverviewLayoutOption.list => SingleChildScrollView(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 12.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisSize: MainAxisSize.min,
+                children: items,
+              ),
+            ),
+          ),
+        _OverviewLayoutOption.grid => SingleChildScrollView(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 12.0),
+              child: Column(
+                children: [
+                  // We use a manual column + wrap approach rather than a
```

Comment:
I would really prefer if we used a gridview here and simply change the column count based on the size of the screen. The only difficult part is the ability to modify the ratio of the items.